samples/user-interface/appwidgets/src/main/java/com/example/platform/ui/appwidgets/glance/weather/WeatherWorker.kt

```diff
@@ -76,17 +76,10 @@ class WeatherWorker(
     }
 
     override suspend fun doWork(): Result {
-        val manager = GlanceAppWidgetManager(context)
-        val glanceIds = manager.getGlanceIds(WeatherGlanceWidget::class.java)
         return try {
-            // Update state to indicate loading
-            setWidgetState(glanceIds, WeatherInfo.Loading)
-            // Update state with new data
-            setWidgetState(glanceIds, WeatherRepo.getWeatherInfo())
-
+            WeatherGlanceWidget().updateAll(context)
```

Comment:
Do we need the worker at all? Should we just relay on the updateMillis mechanism?

The provideContent() will be called when the widget update right?

Reply:
UpdateMilis can't be below 30 minutes?

Reply:
That's correct. Also, it's not a guarantee, system battery restrictions might apply

https://developer.android.com/develop/ui/views/appwidgets/advanced#periodic

Reply:
Okay. I found nothing about this in the documentation, so glad I stumbled upon it here.
Thanks.

Reply:
The javadocs of `updatePeriodMillis` contains that information too
https://developer.android.com/reference/android/appwidget/AppWidgetProviderInfo#updatePeriodMillis

@secondsun maybe it should be highlighted in the Glance docs.

Reply:
Dang, I just looked at that 🙃 
Sorry for siderailing this PR.